### PR TITLE
Replace mono_assembly_name_free use with mono_assembly_name_free_internal.

### PR DESCRIPTION
**Pull request by jaykrell**
Maybe just remove the GC mode change and remove the internal form.
We don't really equate malloc/free with GC/locks/hangs, though arguably we should. Arguably this change is right.

### CODE REVIEW

File: mono/metadata/assembly-internals.h [hunk 1]
```diff
@@ -34,7 +34,14 @@ typedef enum {
 G_ENUM_FUNCTIONS (MonoAssemblyNameEqFlags);
 
 void
-mono_assembly_name_free_internal (MonoAssemblyName *aname);
+mono_assembly_name_free_internal_slow (MonoAssemblyName *aname);
+
+static inline void
+mono_assembly_name_free_internal (MonoAssemblyName *aname)
+{
+	if (aname)
```
Comment (vargaz) [inline mono/metadata/assembly-internals.h]: Why does this has to be inline ?

Comment (jaykrell) [reply mono/metadata/assembly-internals.h]: It does not have to be. I can undo that.

File: mono/metadata/assembly.c [hunk 2]
```diff
@@ -3188,20 +3188,14 @@ mono_assembly_load_from (MonoImage *image, const char *fname,
...
 
 /**
- * mono_assembly_name_free:
+ * mono_assembly_name_free_internal:
  * \param aname assembly name to free
  * 
  * Frees the provided assembly name object.
  * (it does not frees the object itself, only the name members).
+ *
+ * Call mono_assembly_name_free_internal instead.
```
Comment (jaykrell) [inline mono/metadata/assembly.c]: Need to remove comment.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by CoffeeFlux** [comment]
This looks good, just missing an include I think:
```
/mnt/jenkins/workspace/test-mono-pull-request-wasm/mono/mini/mini-wasm-debugger.c:416:2: error: implicit declaration of function 'mono_assembly_name_free_internal' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        mono_assembly_name_free_internal (aname);
        ^
/mnt/jenkins/workspace/test-mono-pull-request-wasm/mono/mini/mini-wasm-debugger.c:416:2: note: did you mean 'mono_assembly_name_free'?
/mnt/jenkins/workspace/test-mono-pull-request-wasm/mono/metadata/assembly.h:116:37: note: 'mono_assembly_name_free' declared here
MONO_API MONO_RT_EXTERNAL_ONLY void mono_assembly_name_free   (MonoAssemblyName *aname);
```

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins squash

**Pull request merged**